yt_dlp/extractor/archiveorg.py

```diff
@@ -945,3 +946,238 @@ def _real_extract(self, url):
+                                        'pv': '4.26.9',
+                                        'dr': '1920x1080',
+                                        'cpl': 'en_US',
+                                        'lc': 'en_US',
+                                        'adi': '%5B%7B%22type%22%3A%22pre%22%2C%22exposure%22%3Afalse%2C%22replayExposure%22%3Afalse%7D%5D',
+                                        'adu': '%2F',
+                                        'videoId': vod_id,
+                                        'cc': user_country,
+                                    }), video_id)
+
+        formats = []
+
+        if len(vod_data.get('streams', [])) > 1:
+            self.report_warning('Multiple streams found. Only the first stream will be downloaded.')
+        stream = traverse_obj(vod_data, ('streams', 0), {})
+
+        max_stream = max(stream.get('videos', []), key=lambda v: traverse_obj(v, ('bitrate', 'video')), default=None)
+        if max_stream is not None:
+            params = {arg.get('name'): arg.get('value') for arg in stream.get('keys', []) if arg.get('type') == 'param'}
+            m3u8_doc = self._download_wbm_page(max_stream.get('source'), video_id, note='Downloading m3u8', query=params, fatal=False)
+            if m3u8_doc:
+                # M3U8 document is not valid, so it needs to be fixed
+                m3u8_doc_lines = m3u8_doc.splitlines()
+                modified_m3u8_doc_lines = []
+                url_base = max_stream.get('source').rsplit('/', 1)[0]
+                first_segment = None
+                for line in m3u8_doc_lines:
+                    if line.startswith('#'):
+                        modified_m3u8_doc_lines.append(line)
+                    else:
+                        modified_line = f'{self._WAYBACK_BASE_URL}{url_base}/{line}?{urllib.parse.urlencode(params)}'
+                        modified_m3u8_doc_lines.append(modified_line)
+                        if first_segment is None:
+                            first_segment = modified_line
+                modified_m3u8_doc = '\n'.join(modified_m3u8_doc_lines)
+
+                # Segments may not have been archied. See 101870
+                first_segment_req = self._request_webpage(HEADRequest(first_segment), video_id, note='Check first segment availablity', errnote=False, fatal=False)
+                if first_segment_req:
+                    formats, _ = self._parse_m3u8_formats_and_subtitles(modified_m3u8_doc, ext='mp4', video_id=video_id)
+
+        # For parts of the project MP4 files were archived
+        max_video = max(traverse_obj(vod_data, ('videos', 'list'), []), key=lambda v: traverse_obj(v, ('bitrate', 'video')), default=None)
+        if max_video is not None:
+            video_url = self._WAYBACK_BASE_URL + max_video.get('source')
+            video_req = self._request_webpage(HEADRequest(video_url), video_id, note='Check video availablity', errnote=False, fatal=False)
```

Comment:
this is same as `_download_wbm_page`, no?

Reply:
Only making a HEAD request to not download the entire video

Reply:
I assume the lack of retries here is also intentional?

Reply:
Hmm, sort of. I added the retries due to the IA servers being quite unreliable and requests often timing out or being aborted. Originally, even 404 requests would be retired, which caused these availability checks to need three requests to fail. I did just adjust `_download_wbm_page` to no longer retry 404ed requests, since that is dumb.

Any suggestions for a nice way to add the option of HEAD requests to `_download_wbm_page`?